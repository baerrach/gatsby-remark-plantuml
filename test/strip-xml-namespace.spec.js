const cheerio = require(`cheerio`)
const testRemarkPlugin = require(`./test-remark-plugin`)

const code = `
\`\`\`plantuml
@startuml

note
  this contains */ which is a html comment
  and will cause MDX to fail to process it
end note

@enduml
\`\`\`
`

describe(`Strip XML namespace`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it(`option defaults to true`, async () => {
    const ast = await testRemarkPlugin.testPlugin({
      code,
    })

    const $ = cheerio.load(ast.children[0].value)

    expect($(`svg`).attr(`xmlns`)).toBeUndefined()
    expect($(`svg`).attr(`link`)).toBeUndefined()
  })

  it(`strips when manually set to true`, async () => {
    const ast = await testRemarkPlugin.testPlugin({
      code,
      options: {
        stripXmlNamespace: true,
      },
    })

    const $ = cheerio.load(ast.children[0].value)

    expect($(`svg`).attr(`xmlns`)).toBeUndefined()
    expect($(`svg`).attr(`link`)).toBeUndefined()
  })

  it(`does not strip when set to false`, async () => {
    const ast = await testRemarkPlugin.testPlugin({
      code,
      options: {
        stripXmlNamespace: false,
      },
    })

    const $ = cheerio.load(ast.children[0].value)

    expect($(`svg`).attr(`xmlns`)).toBe(`http://www.w3.org/2000/svg`)
    expect($(`svg`).attr(`xlink`)).toBe(`http://www.w3.org/1999/xlink`)
  })
})
