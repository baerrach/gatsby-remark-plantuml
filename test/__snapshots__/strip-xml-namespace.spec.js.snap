// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strip XML namespace does not strip when set to false 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 11,
          "offset": 136,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" xlink=\\"http://www.w3.org/1999/xlink\\" contentscripttype=\\"application/ecmascript\\" contentstyletype=\\"text/css\\" height=\\"22px\\" preserveAspectRatio=\\"none\\" style=\\"max-width: 100%; height: auto;\\" version=\\"1.1\\" viewBox=\\"0 0 22 22\\" width=\\"22px\\" zoomAndPan=\\"magnify\\"><defs/><g></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 12,
      "offset": 137,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Strip XML namespace option defaults to true 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 11,
          "offset": 136,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg contentscripttype=\\"application/ecmascript\\" contentstyletype=\\"text/css\\" height=\\"22px\\" preserveAspectRatio=\\"none\\" style=\\"max-width: 100%; height: auto;\\" version=\\"1.1\\" viewBox=\\"0 0 22 22\\" width=\\"22px\\" zoomAndPan=\\"magnify\\"><defs/><g></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 12,
      "offset": 137,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Strip XML namespace strips when manually set to true 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 11,
          "offset": 136,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg contentscripttype=\\"application/ecmascript\\" contentstyletype=\\"text/css\\" height=\\"22px\\" preserveAspectRatio=\\"none\\" style=\\"max-width: 100%; height: auto;\\" version=\\"1.1\\" viewBox=\\"0 0 22 22\\" width=\\"22px\\" zoomAndPan=\\"magnify\\"><defs/><g></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 12,
      "offset": 137,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
